src/Native/System.Native/pal_networking.h

```diff
@@ -117,6 +117,23 @@ struct LingerOption
     int32_t Seconds; // Number of seconds to linger for
 };
 
+struct IOVector
+{
+    uint8_t* Base;
+    uintptr_t Count;
+};
+
+struct MessageHeader
+{
+    uint8_t* SocketAddress;
```

Comment:
AFAIK we're marshalling strings as sequences of UTF-8 code units on *nix, so I think `uint8_t` is still preferable (esp. since UTF-8 code units may span multiple bytes). At boundaries between shim and platform code, however, you may need to cast between `uint8_t*` and `char*` as per the platform signatures. @sokket has some ~~paranoid~~ good code [here](https://github.com/dotnet/corefx/pull/3868/files#diff-a533f7324e00c23d36788e9084520906R33) to ensure the legality of the cast.